runtime/android/core/src/org/xwalk/core/XWalkCoreWrapper.java

```diff
@@ -203,6 +232,39 @@ private boolean checkCoreArchitecture() {
         return true;
     }
 
+    private boolean verifyPackageInfo(PackageInfo packageInfo,
+            String algorithm, String fingerprint) {
+        if (packageInfo.signatures == null) {
+            Log.d(TAG, "No signature in package info");
```

Comment:
I had understood that if we build with the integrity check enabled, a failure in checking the runtime's signature would have to be fatal and we'd need to be as vocal as we can, since it would mean the user has a tampered runtime APK installed.